src/S3/MultipartCopy.php

```diff
@@ -127,16 +128,24 @@ private function createPart($partNumber, $partsCount)
         foreach ($params as $k => $v) {
             $data[$k] = $v;
         }
-
-        list($bucket, $key) = explode('/', ltrim($this->source, '/'), 2);
-        $data['CopySource'] = '/' . $bucket . '/' . implode(
-            '/',
-            array_map(
-                'urlencode',
-                explode('/', rawurldecode($key))
-            )
-        );
+        // The source parameter here is usually a string, but can be overloaded as an array
+        // if the key contains a '?' character to specify where the query parameters start
+        if (is_array($this->source)) {
+            $key = str_replace('?', '%3F', $this->source['source_key']);
```

Comment:
Does this replacement of the `?` need to be composed with the `rawurlencode` and `urlencode` calls?

Reply:
I fixed it to call rawurlencode instead